corehq/apps/locations/models.py

```diff
@@ -389,10 +387,24 @@ def save(self, *args, **kwargs):
         if not self.location_id:
             self.location_id = uuid.uuid4().hex
         set_site_code_if_needed(self)
-        self.supply_point_id = sync_supply_point(self)
+        sync_supply_point(self)
         super(SQLLocation, self).save(*args, **kwargs)
         publish_location_saved(self.domain, self.location_id)
 
+    def delete(self, *args, **kwargs):
```

Comment:
When we built that out, we were thinking of it as analogous to deleting a user.  The associated cases still exist, and you can dig them up, but they're hidden from normal views.  There had been an archival workflow around locations for a while, but that was confusing, since people would frequently make mistakes in setting up their project, then try to archive things and start over.  Archiving only hides a location, so it prevents you from making another location with the same site code, or with the same name and parent.